src/gwt/src/org/rstudio/studio/client/workbench/views/source/Source.java

```diff
@@ -731,6 +708,15 @@ else if (uiPrefs_.enableEmacsKeybindings().getGlobalValue())
       
       // handle chunk options event
       handleChunkOptionsEvent();
+      
+      consoleEditorProvider_.getConsoleEditor().addFocusHandler(new FocusHandler()
```

Comment:
Do we have any guarantee that the `Console` is constructed eagerly enough (ie, before we construct `Source`) that we won't run into race conditions here? Note that the `consoleEditorProvider_` doesn't get initialized until here: https://github.com/rstudio/rstudio/blob/master/src/gwt/src/org/rstudio/studio/client/workbench/views/console/shell/Shell.java#L114